streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -85,14 +86,55 @@ public void timeDifferenceMustNotBeNegative() {
         JoinWindows.of(-1);
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void afterBelowLower() {
-        JoinWindows.of(anySize).after(-anySize - 1);
+        final JoinWindows windowSpec = JoinWindows.of(anySize);
+        try {
+            windowSpec.after(-anySize - 1);
+            fail("window end time should not be before window start time");
+        } catch (final IllegalArgumentException e) {
+            // expected
+        }
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void beforeOverUpper() {
-        JoinWindows.of(anySize).before(-anySize - 1);
+        final JoinWindows windowSpec = JoinWindows.of(anySize);
+        try {
+            windowSpec.before(-anySize - 1);
+            fail("window start time should not be after window end time");
+        } catch (final IllegalArgumentException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void testUntil() {
+        final JoinWindows windowSpec = JoinWindows.of(anySize);
+        final long windowSize = windowSpec.size();
+        assertEquals(windowSize, windowSpec.until(windowSize).maintainMs());
+    }
+
+    @Test
+    public void testMaintainMsForLargeWindowSiye() {
```

Comment:
Again i feel there is a more descriptive name for this test. 
sth like: `shouldUseWindowSizeForMaintainMsWhenSizeLargerThanDefaultMaintainMs`